src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -333,9 +333,15 @@ Copyright (c) .NET Foundation. All rights reserved.
         SuppressNotFoundError="true">
       <Output TaskParameter="FullFrameworkReferenceAssemblyPaths" PropertyName="_FullFrameworkReferenceAssemblyPaths"/>
     </GetReferenceAssemblyPaths>
+
+    <PropertyGroup>
+      <_PackageReferenceList>@(PackageReference)</_PackageReferenceList>
```

Comment:
I think the way to do this is something like this:

```xml
<_ExistingReferenceAssembliesPackageReference Include="@(PackageReference)" Condition="'%(PackageReference.Identity)' == 'Microsoft.NETFramework.ReferenceAssemblies'"/>

<PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="$(MicrosoftNETFrameworkReferenceAssembliesLatestPackageVersion)" IsImplicitlyDefined="true"
                  Condition="'$(_FullFrameworkReferenceAssemblyPaths)' == '' and '$(_ExistingReferenceAssembliesPackageReference)' == ''"/>
```

That way you're filtering the items and checking if there are any that match instead of concatenating them all and doing a string search.

Reply:
That's much cleaner, thanks. I've updated this PR.